### PR TITLE
CSS naming convention docs

### LINKED ISSUE #1454: Document naming conventions for CSS classes
We should write down some CSS naming conventions to make sure we all write CSS class names in the same way and allow contributors and theme developers to know about them.

I think we should stick to WooCommerce naming convention (https://github.com/woocommerce/woocommerce/wiki/CSS-SASS-coding-guidelines-and-naming-conventions) but stick to the prefix we are currently using (`.wc-block-`).

Thoughts?

### PR DESCRIPTION
**Pull request by mikejolley**
This PR adds some docs for CSS naming conventions within the blocks project.

Closes #1454

Since this doc needed a home I've done a small amount of organising too:

- GITHUB specific MD files have been moved to `./github` to keep the root directory clean.
- I've added missing CODE OF CONDUCT (duplicated from woo core) and SECURITY (linking to hacker1) to this directory too.
- Added `/docs` to house development docs. I moved the release MD file there also.

`docs` is excluded from builds.

### CODE REVIEW

File: docs/coding-guidelines.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+#### Prefixing
+
+As a WordPress plugin, Blocks has to play nicely with other plugins and themes, and WordPress itself. To minimize conflict potential all classes should be prefixed with `.wc-block-`.
+
+#### Naming
+
+As well as using a `wc-block-` prefix, all class names assigned to an element must be prefixed with the name of the package, followed by a dash and the name of the directory in which the component resides. Any descendent of the component's root element must append a dash-delimited descriptor, separated from the base by two consecutive underscores \_\_.
+
+    Root element: wc-block-package-directory
```
Comment (haszari) [inline docs/coding-guidelines.md]: Is this equivalent to `Block` in BEM? If so might be clearer to mention that.

File: docs/coding-guidelines.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    Child elements: wc-block-package-directory__descriptor-foo-bar
```
Comment (haszari) [inline docs/coding-guidelines.md]: And this is BEM's `Element`?

File: docs/coding-guidelines.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+For example, `assets/base/components/checkbox-list` uses the class name: `wc-block-checkbox-list`.
+
+The **root element** is considered to be the highest ancestor element returned by the default export in the index.js. Notably, if your folder contains multiple files, each with their own default exported component, only the element rendered by that of index.js can be considered the root. All others should be treated as **descendents**.
```
Comment (haszari) [inline docs/coding-guidelines.md]: Great to have this spelled out clearly, this confused me at first. Again, this is nicely consistent with BEM's `Block` as I understand it :)

File: docs/coding-guidelines.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+Naming is not strictly tied to the DOM so it **doesn’t matter how many nested levels deep a descendent element is**. The naming convention is there to help you identify relationships with the root element.
+
+**Examples:**
+
+-   `wc-block-dropdown-selector` (root element)
+-   ├── `wc-block-dropdown-selector__input` (descendent)
+-   ├── `wc-block-dropdown-selector__input--active` (modifier on descendent)
```
Comment (haszari) [inline docs/coding-guidelines.md]: Ah! BEM `Modifier` example! Nice, I hadn't come across this in the code yet :)

File: docs/coding-guidelines.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+	text-align: left;
+	/*rtl:end:ignore*/
+}
+```
+
+### SCSS File Naming Conventions for Blocks
+
+The build process will split SCSS from within the blocks library directory into two separate CSS files when Webpack runs.
+
+Styles placed in a `style.scss` file will be built into `build/style.css`, to load on the front end theme as well as in the editor. If you need additional styles specific to the block's display in the editor, add them to an `editor.scss`.
```
Comment (haszari) [inline docs/coding-guidelines.md]: This is super handy useful info :)

File: docs/releasing-new-versions.md [hunk 6]
```diff
@@ -0,0 +1,168 @@
+# Releasing New Versions
```
Comment (haszari) [inline docs/releasing-new-versions.md]: FYI @nerrad has some changes to this doc in #1655

File: docs/coding-guidelines.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+To avoid class name collisions, class names must adhere to the following guidelines, based on the [BEM methodology](https://en.bem.info/methodology/) and [Gutenberg coding standards](https://developer.wordpress.org/block-editor/contributors/develop/coding-guidelines/).
+
+#### Prefixing
+
+As a WordPress plugin, Blocks has to play nicely with other plugins and themes, and WordPress itself. To minimize conflict potential all classes should be prefixed with `.wc-block-`.
+
+#### Naming
+
+As well as using a `wc-block-` prefix, all class names assigned to an element must be prefixed with the name of the package, followed by a dash and the name of the directory in which the component resides. Any descendent of the component's root element must append a dash-delimited descriptor, separated from the base by two consecutive underscores \_\_.
```
Comment (haszari) [inline docs/coding-guidelines.md]: What does "package" mean in this context? Could be helpful to explicitly define it, or include an example with "package" (or both).

File: docs/coding-guidelines.md (hunk 1 above)
Comment (Aljullu) [inline docs/coding-guidelines.md]: This seems to render as code in the rich view:



Was that in purpose?

File: docs/coding-guidelines.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+Naming is not strictly tied to the DOM so it **doesn’t matter how many nested levels deep a descendent element is**. The naming convention is there to help you identify relationships with the root element.
+
+**Examples:**
+
+-   `wc-block-dropdown-selector` (root element)
+-   ├── `wc-block-dropdown-selector__input` (descendent)
+-   ├── `wc-block-dropdown-selector__input--active` (modifier on descendent)
+-   └── `wc-block-dropdown-selector__placeholder` (descendent)
+
+### RTL Styles
```
Comment (Aljullu) [inline docs/coding-guidelines.md]: Great you added this section and `SCSS File Naming Conventions for Blocks`! :clap:

File: docs/coding-guidelines.md (hunk 7 above)
Comment (Aljullu) [inline docs/coding-guidelines.md]: Nitpicking suggestion but `\_\_` was a bit confusing for me when reviewing the raw view. What would you think about wrapping it with accents as follows?

```suggestion
As well as using a `wc-block-` prefix, all class names assigned to an element must be prefixed with the name of the package, followed by a dash and the name of the directory in which the component resides. Any descendent of the component's root element must append a dash-delimited descriptor, separated from the base by two consecutive underscores `__`.
```

File: docs/coding-guidelines.md [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+
+## CSS
+
+### CSS Class Names
+
+To avoid class name collisions, class names must adhere to the following guidelines, based on the [BEM methodology](https://en.bem.info/methodology/) and [Gutenberg coding standards](https://developer.wordpress.org/block-editor/contributors/develop/coding-guidelines/).
+
+#### Prefixing
+
+As a WordPress plugin, Blocks has to play nicely with other plugins and themes, and WordPress itself. To minimize conflict potential all classes should be prefixed with `.wc-block-`.
```
Comment (Aljullu) [inline docs/coding-guidelines.md]: Question from a non-native speaker "To minimize conflict potential" shouldn't be "To minimize potential conflict"? :thinking: 

In addition, I remember @nerrad and @senadir suggested using `.wc-component` for components, I think this doesn't need to be discussed here and can be tackled in the future, but I wanted to point it out so everybody is aware of this.

File: docs/coding-guidelines.md (hunk 3 above)
Comment (Aljullu) [inline docs/coding-guidelines.md]: Another English question: `descendents` or `descendants`?

File: docs/coding-guidelines.md (hunk 1 above)
Comment (mikejolley) [reply docs/coding-guidelines.md]: I'm rewriting this part, but yes I'm trying to make examples stand out from the descriptions.

File: docs/releasing-new-versions.md (hunk 6 above)
Comment (mikejolley) [reply docs/releasing-new-versions.md]: 👍 Will need to resolve merge conflicts depending on whats first :)

File: docs/coding-guidelines.md (hunk 3 above)
Comment (mikejolley) [reply docs/coding-guidelines.md]: Beats me, maybe you're right.

**Comment by mikejolley** [comment]
Rebased with the changes from @nerrad and resolved all feedback. Merging this in :)

**Pull request merged**